assisted-events-scrape/tests/integration/test_integration.py

```diff
@@ -0,0 +1,34 @@
+from config import ElasticsearchConfig
+import elasticsearch
+import waiting
+import time
+
+INIT_DELAY_SECONDS = 10
+ALL_EVENTS_NUMBER = 101
+
+
+class TestIntegration:
+    def setup(self):
+        self._wait_for_elastic()
+
+    def test_all_docs(self):
+        n_documents = self._es_client.count(index=self._config.index)['count']
+        assert(n_documents == ALL_EVENTS_NUMBER)
+
+    def _wait_for_elastic(self):
+        waiting.wait(
+            lambda: self._init_elastic_client(),
+            timeout_seconds=300,
+            sleep_seconds=5,
+            waiting_for="Wait for elasticsearch",
+            expected_exceptions=Exception,
+        )
+
+    def _init_elastic_client(self) -> bool:
+        self._config = ElasticsearchConfig.create_from_env()
+        self._es_client = elasticsearch.Elasticsearch(self._config.host)
+        is_elastic_ready = self._es_client.indices.exists(index=self._config.index)
+        if is_elastic_ready:
+            # As elastic is eventually consistent, we need to make sure we wait more than refresh rate
+            time.sleep(INIT_DELAY_SECONDS)
```

Comment:
can we instead wait for 3 consecutive succeeding requests to ES?
I'd rather not sleep if possible

Reply:
The issue is that if we don't sleep, 3 request might happen very quickly and not give time to elastic to sync.
What we can do (in the future), is enable writing synchronously when the application is run in test mode. This would only affect performance.
To mitigate this, we could read "infinite" times during INIT_DELAY_SECONDS and if the result is eventually positive, we can return. Not sure how much would this help though. WDYT?

Reply:
The part that I don't like with static sleeping is not knowing the state after those X seconds. If we have checks integrated with sleeps then we can know when X seconds are becoming not enough

Not sure in this case what would make stability more guaranteed (as you said, it's not easy understanding that a service is fully stable), but we don't need to fix it in this PR's scope